block_model.rb

```diff
@@ -0,0 +1,73 @@
+class BlockModel
+  attr_reader :blocks
+
+  def initialize(blocks)
+    @blocks = blocks
+  end
+
+  def reblock(rx, ry, rz)
+    new_blocks = []
+    nx = @blocks.length / rx
+    ny = @blocks.length[0] / ry
+    nz = @blocks.length[0][0] / rz
+    for i in (0..nx)
+      new_blocks[i] = []
+      for j in (0..ny)
+        new_blocks[i][j] = []
+        for k in (0..nz)
+          new_blocks[i][j][k] = {}
+        end
+      end
+    end
+
+    n_i = 0
+    n_j = 0
+    n_k = 0
```

Comment:
Through this file we can see variables like this ones, that don't say anything in terms of functionality. In order to make this code easier to understand, all variables should be self explanatory.